File: MATLAB/+cfg/default_cfg.m
function s = default_cfg()
%DEFAULT_CFG  Central place for run-time policy (no hidden defaults in tasks).
addpath(fullfile(fileparts(fileparts(mfilename('fullpath'))), 'src', 'utils'));
p = project_paths();
s = struct();
s.dataset_id = '';        % must be set by caller
s.method     = '';        % must be set by caller
s.imu_file   = '';
s.gnss_file  = '';
s.truth_file = '';

s.paths = p;

s.plots = struct( ...
    'popup_figures', false, ...
    'save_pdf',      false, ...
    'save_png',      false);
end

File: PYTHON/src/cfg/default_cfg.py
"""Default configuration stub mirroring MATLAB ``cfg.default_cfg``."""
from __future__ import annotations

from .. import project_paths  # type: ignore


def default_cfg() -> dict:
    """Return default configuration dictionary."""
    paths = project_paths.project_paths() if hasattr(project_paths, "project_paths") else {}
    return {
        "dataset_id": "",
        "method": "",
        "imu_file": "",
        "gnss_file": "",
        "truth_file": "",
        "paths": paths,
        "plots": {
            "popup_figures": False,
            "save_png": True,
        },
        "zupt": {
            "acc_movstd_thresh": 0.15,
            "min_pre_lift_s": 5,
            "speed_thresh_mps": 0.30,
        },
    }
